### PR TITLE
first cut at user profile library cards

### PR DESCRIPTION
**Pull request by 2is10**
@yipingliao

### CODE REVIEW

**Comment by yipingliao** [comment]
Yay! Will definitely look through this in detail.

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfileLibrariesList.tpl.html [hunk 1]
```diff
@@ -1 +1,21 @@
...
+        <div class="kf-upl-card-name-clip">
+          {{lib.name}}
+        </div>
+      </div>
+    </div>
+    <div class="kf-upl-counts">
+      <ng-pluralize class="kf-upl-count" count="lib.numKeeps" when="{'1':'{} Keep','other':'{} Keeps'}" ng-if="lib.numKeeps > 0"></ng-pluralize>
+      <ng-pluralize class="kf-upl-count" count="lib.numFollowers" when="{'1':'{} Follower','other':'{} Followers'}" ng-if="lib.numFollowers > 0"></ng-pluralize>
+    </div>
+    <div class="kf-upl-card-byline">{{lib.owner.firstName}} {{lib.owner.lastName}}</div>
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfileLibrariesList.tpl.html:17]: I am surprised to see that the owner's name is nested within the library's information and when clicked, will go to the library. I was expecting that the owner's name will be with the owner's picture and when clicked will go to the owner's user profile, like the picture. This kind of grouping will also be easier for tagging microdata.

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfileActionService.js [hunk 2]
```diff
@@ -5,21 +5,27 @@ angular.module('kifi')
 .factory('userProfileActionService', [
   '$http', '$q', 'routeService', 'Clutch',
   function ($http, $q, routeService, Clutch) {
-    var clutchParamsRecos = {
+    var clutchParams = {
       cacheDuration: 2000
     };
 
+    function getData(res) {
+      return res.data;
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfileActionService.js:13]: Do you think it's okay to have removed the null check for `res`?

**Comment by yipingliao** [comment]
Wow. I'm looking forward to seeing these library cards!

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfileLibrariesList.tpl.html (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfileLibrariesList.tpl.html:17]: Yeah, I wasn’t sure how it should work. Maybe the whole author area should link to the author’s profile, but that’s also the area that shows the library’s cover image. Or maybe just the picture and name. For now I just made it the picture until I can have a conversation with Mark. Easy to change.

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfileActionService.js (hunk 2 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfileActionService.js:13]: Yes. From `$http` [doc](https://docs.angularjs.org/api/ng/service/$http#usage):

> Returns a `promise` object with the standard `then` method and two http specific methods: `success` and `error`. The `then` method takes two arguments a success and an error callback which will be called with a response object. ...

**Comment by 2is10** [comment]
Thanks for reviewing! Still a lot to do.

File: kifi-backend/modules/shoebox/angular/src/userProfile/userProfileLibrariesList.tpl.html (hunk 1 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/userProfile/userProfileLibrariesList.tpl.html:17]: Ah, I see.

**Pull request merged**